app/src/org/commcare/dalvik/activities/EntitySelectActivity.java

```diff
@@ -785,6 +786,7 @@ private void select() {
         // create intent for return and store path
         Intent i = new Intent(EntitySelectActivity.this.getIntent());
         i.putExtra(SessionFrame.STATE_DATUM_VAL, selectedIntent.getStringExtra(SessionFrame.STATE_DATUM_VAL));
+        ACRAUtil.addCustomData(ACRAUtil.CURRENT_CASE, selectedIntent.getStringExtra(SessionFrame.STATE_DATUM_VAL));

```

Comment:
Yeah, we host it on Cloudant. Maybe we should sit down at some point and go over everything?